apps/heft/src/cli/HeftToolsCommandLineParser.ts

```diff
@@ -229,6 +229,9 @@ export class HeftToolsCommandLineParser extends CommandLineParser {
     if (currentCwd !== buildFolder) {
       // Update the CWD to the project's build root. Some tools, like Jest, use process.cwd()
       this._terminal.writeVerboseLine(`CWD is "${currentCwd}". Normalizing to project build folder.`);
+      if (FileSystem.existsAsync('.heft')) {
+        process.chdir('.heft');
+      }
```

Comment:
This won't work the way you expect. `FileSystem.existsAsync` will always return truthy because it returns a promise, even if that promise resolves to `false`. Once that gets fixed, this won't correct the path if a) the current folder isn't already the project root (i.e. - someone is invoking Heft from `/my-project/src`) or if it's the first time Heft has been run on a fresh clone and the `.heft` folder doesn't exist.

The `.heft` folder's absolute path is also already exposed as `this._heftConfiguration.projectHeftDataFolder`.

Reply:
My recommendation would be to make this function `async` and change this to:

```suggestion
      await FileSystem.ensureFolderAsync(this._heftConfiguration.projectHeftDataFolder);
      process.chdir(this._heftConfiguration.projectHeftDataFolder);
```

Reply:
As an aside - is there a performance hit for changing the cwd twice? What does `chdir` do under the hood?

Reply:
@dmichon-msft Any idea how this behavior would change the performance?